integration-tests/bats/fetch.bats

```diff
@@ -0,0 +1,322 @@
+#!/usr/bin/env bats
+load $BATS_TEST_DIRNAME/helper/common.bash
+
+setup() {
+    setup_common
+    TMPDIRS=$(pwd)/tmpdirs
+    mkdir -p $TMPDIRS/{rem1,repo1}
+
+    # repo1 -> rem1 -> repo2
+    cd $TMPDIRS/repo1
+    dolt init
+    dolt remote add origin file://../rem1
+    dolt remote add test-remote file://../rem1
+    dolt push origin main
+
+    cd $TMPDIRS
+    dolt clone file://rem1 repo2
+    cd $TMPDIRS/repo2
+    dolt branch feature
+    dolt remote add test-remote file://../rem1
```

Comment:
This remote should point to a remote that has different content. Then in tests which pull from `test-remote`, you can verify that you didn't fetch from origin.